code/modules/clothing/spacesuits/rig/suits/station.dm

```diff
@@ -158,9 +159,10 @@
 	suit_type = "hazmat hardsuit"
 	desc = "An Anomalous Material Interaction hardsuit that protects against the strangest energies the universe can throw at it."
 	icon_state = "science_rig"
-	armor = list(melee = 45, bullet = 5, laser = 45, energy = 80, bomb = 60, bio = 100, rad = 100)
+	armor = list(melee = 45, bullet = 5, laser = 40, energy = 65, bomb = 60, bio = 100, rad = 100)
 	slowdown = 1
 	offline_vision_restriction = 1
+	emp_protection = 40
```

Comment:
It seems pretty well added. Tested EMP's in game on the upgraded RD and industrial suit. Industrial suit took more module damage, both took same battery damage. I could fix this by factoring it in here
/obj/item/weapon/rig/proc/shock(mob/user)
	if (electrocute_mob(user, cell, src)) //electrocute_mob() handles removing charge from the cell, no need to do that here.
		spark_system.start()
		if(user.stunned)
			return 1
	return 0
Otherwise the rest seems fine.